src/client.ts

```diff
@@ -0,0 +1,620 @@
+    this.makeCall('getMemoryPoolContent')
+
+  getMemoryPoolInformation = (): Promise<MempoolInfo> =>
+    this.makeCall('getMemoryPoolInformation')
+
+  getMempoolAncestors = (
+    txid: string,
+    verbose?: boolean
+  ): Promise<MempoolContent[] | string[] | null[]> =>
+    this.makeCall('getMempoolAncestors', txid, verbose)
+
+  getMempoolDescendants = (
+    txid: string,
+    verbose?: boolean
+  ): Promise<MempoolContent[] | string[] | null[]> =>
+    this.makeCall('getMempoolDescendants', txid, verbose)
+
+  getMempoolEntry = (txid: string): Promise<MempoolContent> =>
+    this.makeCall('getMempoolEntry', txid)
+
+  getMempoolInfo = (): Promise<MempoolInfo> => this.makeCall('getMempoolInfo')
+
+  getMiningInfo = (): Promise<MiningInfo> => this.makeCall('getMiningInfo')
+
+  getNetTotals = (): Promise<NetTotals> => this.makeCall('getNetTotals')
+
+  getNetworkHashPs = (nblocks = '120', height?: number): Promise<number> =>
+    this.makeCall('getNetworkHashPs', nblocks, height)
+
+  getNetworkInfo = (): Promise<NetworkInfo> => this.makeCall('getNetworkInfo')
+
+  getNewAddress = (
+    label = '',
+    addressType?: 'legacy' | 'bech32' | 'p2sh-segwit'
+  ): Promise<string> => this.makeCall('getNewAddress', label, addressType)
+
+  getPeerInfo = (): Promise<PeerInfo[]> => this.makeCall('getPeerInfo')
+
+  getRawChangeAddress = (): Promise<string> =>
+    this.makeCall('getRawChangeAddress')
+
+  getRawMempool = (
+    verbose?: boolean
+  ): Promise<MempoolContent[] | string[] | null[]> =>
+    this.makeCall('getRawMempool', verbose)
+
+  getRawTransaction = (
+    txid: string,
+    verbose?: boolean
+  ): Promise<FetchedRawTransaction | string> =>
+    this.makeCall('getRawTransaction', txid, verbose)
+
+  getReceivedByAddress = (address: string, minconf?: number): Promise<number> =>
+    this.makeCall('getReceivedByAddress', address, minconf)
+
+  getTransaction = (
+    txid: string,
+    include_watchonly?: boolean
+  ): Promise<WalletTransaction> =>
+    this.makeCall('getTransaction', txid, include_watchonly)
+
+  getTxOut = (
+    txid: string,
+    index: number,
+    include_mempool?: boolean
+  ): Promise<TxOutInBlock> =>
+    this.makeCall('getTxOut', txid, index, include_mempool)
+
+  getTxOutProof = (txids: string[], blockhash?: string): Promise<string> =>
+    this.makeCall('getTxOutProof', txids, blockhash)
+
+  getTxOutSetInfo = (): Promise<UTXOStats> => this.makeCall('getTxOutSetInfo')
+
+  getUnconfirmedBalance = (): Promise<number> =>
+    this.makeCall('getUnconfirmedBalance')
+
+  getWalletInfo = (): Promise<WalletInfo> => this.makeCall('getWalletInfo')
+
+  importAddress = (
+    address: string,
+    label = '',
+    rescan = true,
+    p2sh?: boolean
+  ): Promise<void> =>
+    this.makeCall('importAddress', address, label, rescan, p2sh)
+
+  importMulti = (
+    requests: ImportMultiRequest[],
+    options?: { rescan?: boolean }
+  ): Promise<
+    { success: boolean; error?: { code: string; message: string } }[]
```

Comment:
`{ success: true} | { success: false; error: { code: string; message: string } }`
a little more verbose, maybe worth a `Failable<>` alias if that kind of message is used elsewhere

Reply:
Yes that's better. Added warning to the success part after checking the doc https://bitcoincore.org/en/doc/0.19.0/rpc/wallet/importmulti/